Fatras/include/ActsFatras/Kernel/InteractionList.hpp

```diff
@@ -0,0 +1,344 @@
+//
+// Whether an element is part of the subset is defined by a configurable
+// selector template struct. It must take the element type as its only template
+// parameter and must provide a static `Value` member value. If the value
+// evaluates to `true`, then the corresponding index will be part of the index
+// sequence.
+//
+// Example: The tuple contains four elements, where all but the third one (i=2)
+// will be selected. This should lead to the following recursive expansion
+// where the index sequence of the subset is filled from the front.
+//
+//        TupleIndicesSubsetImpl<..., kCounter=4>          // select index=3
+//     -> TupleIndicesSubsetImpl<..., kCounter=3, 3>       // skip   index=2
+//     -> TupleIndicesSubsetImpl<..., kCounter=2, 3>       // select index=1
+//     -> TupleIndicesSubsetImpl<..., kCounter=1, 1, 3>    // select index=0
+//     -> TupleIndicesSubsetImpl<..., kCounter=0, 0, 1, 3> // terminate
+//
+template <template <typename> typename selector_t, typename tuple_t,
+          size_t kCounter, size_t... kIndices>
+struct TupleSubsetIndicesImpl {
+  static constexpr auto kIndex = kCounter - 1u;
+  static constexpr bool elementSelection =
+      selector_t<std::tuple_element_t<kIndex, tuple_t>>::Value;
+  // recursive type if the element would be selected
+  using SelectElement =
+      typename TupleSubsetIndicesImpl<selector_t, tuple_t, kIndex, kIndex,
+                                      kIndices...>::Type;
+  // recursive type if the element would be skipped
+  using SkipElement =
+      typename TupleSubsetIndicesImpl<selector_t, tuple_t, kIndex,
+                                      kIndices...>::Type;
+  // select recursive type based on the selector decision
+  using Type = std::conditional_t<elementSelection, SelectElement, SkipElement>;
+};
+template <template <typename> typename selector_t, typename tuple_t,
+          size_t... kIndices>
+struct TupleSubsetIndicesImpl<selector_t, tuple_t, 0u, kIndices...> {
+  using Type = std::index_sequence<kIndices...>;
+};
+template <template <typename> typename selector_t, typename tuple_t>
+using TupleSubsetIndices =
+    typename TupleSubsetIndicesImpl<selector_t, tuple_t,
+                                    std::tuple_size_v<tuple_t>>::Type;
+
+/// Check if the given type is a point-like process.
+///
+/// Only checks for the existence of the `generatePathLimits` method using the
+/// method described in https://stackoverflow.com/a/257382 .
+template <typename process_t>
+class IsPointLikeProcess {
+  using OneByte = char;
+  struct TwoByte {
+    char x[2];
+  };
+
+  template <typename T>
+  static OneByte test(decltype(&T::generatePathLimits));
+  template <typename T>
+  static TwoByte test(...);
+
+ public:
+  static constexpr bool Value = (sizeof(test<process_t>(0)) == sizeof(OneByte));
+};
+template <typename process_t>
+struct IsContinuousProcess {
+  static constexpr bool Value = not IsPointLikeProcess<process_t>::Value;
+};
```

Comment:
I think this could be done using the type traits logic that I added for our custom concepts, and might be more concise.